internal/am/job_tracker.go

```diff
@@ -143,18 +147,23 @@ func (jt *JobTracker) saveTasks(ctx context.Context, jobs []amclient.Job) (int,
 			return 0, err
 		}
 		task.WorkflowUUID = jt.workflowUUID
+		tasks = append(tasks, task)
+		savedIDs = append(savedIDs, job.ID)
+	}
 
-		err = jt.ingestsvc.CreateTask(ctx, task)
-		if err != nil {
-			return 0, err
-		}
+	if len(tasks) == 0 {
+		return 0, nil
+	}
 
-		// Add this job ID to the list of savedIDs.
-		jt.savedIDs[job.ID] = struct{}{}
-		count++
+	if err := jt.ingestsvc.CreateTasks(ctx, persistence.TaskSequence(slices.Values(tasks))); err != nil {
```

Comment:
Again, I don't understand the benefit to passing a value from an in-memory slice to an iterator, then saving the `&datatype.Task()` to an array after yielding it back to the caller. :confused: